src/index/walletfilterindex.cpp

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2018-2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <map>
+#include <cmath>
+
+#include <dbwrapper.h>
+#include <hash.h>
+#include <index/walletfilterindex.h>
+#include <node/blockstorage.h>
+#include <util/system.h>
+#include <validation.h>
+
+std::unique_ptr<WalletFilterIndex> g_wallet_filter_index;
+
+using node::UndoReadFromDisk;
+
+constexpr uint8_t DB_BLOCK_HASH{'s'};
+constexpr uint8_t DB_FILTER_POS{'P'};
+constexpr uint8_t DB_SIPHASH_K0{'k'};
+constexpr uint8_t DB_SIPHASH_K1{'K'};
+
+constexpr unsigned int MAX_FLTR_FILE_SIZE = 0x1000000; // 16 MiB
+/** The pre-allocation chunk size for fltr?????.dat files */
+constexpr unsigned int FLTR_FILE_CHUNK_SIZE = 0x100000; // 1 MiB
+
+// value for M explained https://github.com/sipa/writeups/tree/main/minimizing-golomb-filters
+constexpr uint8_t WALLET_FILTER_P = 19;
+constexpr uint32_t WALLET_FILTER_M = 784931; // 1.497137 * pow(2,WALLET_FILTER_P);
```

Comment:
It might make sense to change these values from the BASIC filter values, changing P =32 improves the false positive rate from 1 in 784,931 to 1 in 6,430,154,452 for a trade off of 68% larger filters.

It also might make sense to make it a configuration option when the filters are generated, the value could easily be stored in the index leveldb instead of being a constant.